buildSrc/src/main/kotlin/dokka-conventions.gradle.kts

```diff
@@ -18,62 +11,24 @@ dependencies {
     dokkaPlugin(provider { extens.getByType<VersionCatalogsExtension>().named("libs").findLibrary("dokka.pathsaver").get().get() })
 }
 
-tasks.withType<DokkaTaskPartial>().named("dokkaHtmlPartial") {
-    outputDirectory.set(file("build/dokka"))
-
-
-    pluginsMapConfiguration.put("org.jetbrains.dokka.base.DokkaBase", """{ "templatesDir": "${rootDir.resolve("dokka-templates").canonicalPath.replace('\\', '/')}" }""")
-
-    dokkaSourceSets {
-        configureEach {
-            includes.from(rootDir.resolve("dokka/moduledoc.md").path)
-
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization(\$|\\.).*")
-                reportUndocumented.set(true)
-                skipDeprecated.set(true)
-            }
-
-            // Internal API
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization.internal(\$|\\.).*")
-                suppress.set(true)
-            }
-
-            // Internal JSON API
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization.json.internal(\$|\\.).*")
-                suppress.set(true)
-                reportUndocumented.set(false)
-            }
-
-            // Workaround for typealias
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.serialization.protobuf.internal(\$|\\.).*")
-                suppress.set(true)
-                reportUndocumented.set(false)
-            }
-
-            // Deprecated migrations
-            perPackageOption {
-                matchingRegex.set("kotlinx\\.protobuf(\$|\\.).*")
-                reportUndocumented.set(true)
-                skipDeprecated.set(true)
-            }
+dokka {
+    pluginsConfiguration.html {
+        templatesDir = rootDir.resolve("dokka-templates")
+    }
 
-            // Deprecated migrations
-            perPackageOption {
-                matchingRegex.set("org\\.jetbrains\\.kotlinx\\.serialization\\.config(\$|\\.).*")
-                reportUndocumented.set(false)
-                skipDeprecated.set(true)
-            }
+    dokkaSourceSets.configureEach {
+        includes.from(rootDir.resolve("dokka/moduledoc.md").path)
+        reportUndocumented = true
+        skipDeprecated = true
 
-            sourceLink {
-                localDirectory.set(rootDir)
+        perPackageOption {
+            matchingRegex = ".*\\.internal(\\..*)?"
+            suppress = true
```

Comment:
reportUndocumented = false ?

Reply:
There is no need, as suppressed packages are skipped during analysis, and so no warnings will be present for them